math/arpack/Portfile

```diff
@@ -29,6 +29,29 @@ mpi.setup           require_fortran
 use_autoreconf      yes
 configure.args      home=${worksrcpath} --disable-mpi
 
+# add missing arch flags to one Makefile
+patchfiles-append   patch-Makefile.am.diff
+if {![variant_isset universal]} {
+    post-configure {
+        if {${build_arch} eq "x86_64" || ${build_arch} eq "ppc64"} {
+            set arch_flag -m64
+        } else {
+            set arch_flag -m32
+        }
+        reinplace \
+            "s|@MACPORTS_ARCH_FLAGS@|${arch_flag}|g" \
+            ${worksrcpath}/PARPACK/SRC/MPI/Makefile
+    }
+} else {
+    post-configure {
+        foreach arch ${configure.universal_archs} {
+            reinplace \
+                "s|@MACPORTS_ARCH_FLAGS@|[muniversal_get_arch_flag ${arch} fortran]|g" \
```

Comment:
isn't this going to fail after the first iteration (since `@MACPORTS_ARCH_FLAGS@` has already been replaced)?

Reply:
I do not believe so because each replacement is applied to a different file
(`${worksrcpath}-${arch}/PARPACK/SRC/MPI/Makefile`).

Reply:
I do not believe so because each replacement is applied to a different file
(`${worksrcpath}-${arch}/PARPACK/SRC/MPI/Makefile`).

Reply:
Aha. That makes sense.